tests/unit/models/gpflow/test_models.py

```diff
@@ -1626,3 +1628,331 @@ def test_sparse_variational_pairwise_covariance_for_non_whitened(
     actual_covariance = model.covariance_between_points(query_points_1, query_points_2)
 
     np.testing.assert_allclose(expected_covariance, actual_covariance, atol=1e-4)
+
+
+def multifidelity_autoregressive_nd_dataset(n_dims: int = 1) -> Dataset:
```

Comment:
does it make sense to have those tests here and not in `test_multifidelity_models`? (my guess is it does, but just double-checking)